src/Ens.ts

```diff
@@ -171,101 +262,38 @@ export default class Ens extends EthereumNamingService {
 
-  private async getTextRecord(domain, key): Promise<string | undefined> {
-    const nodeHash = this.namehash(domain);
-    const resolver = await this.getResolverContract(domain);
-    return await this.callMethod(resolver, 'text', [nodeHash, key]);
-  }
-
   private async getResolverContract(
     domain: string,
     coinType?: string,
   ): Promise<Contract> {
     const resolverAddress = await this.resolver(domain);
-    return this.buildContract(
+    return new Contract(
       resolverInterface(resolverAddress, coinType),
       resolverAddress,
+      this.provider
     );
   }
 
-  /**
-   * This was done to make automated tests more configurable
-   */
-  private resolverCallToName(resolverContract: Contract, nodeHash) {
-    return this.callMethod(resolverContract, 'name', [nodeHash]);
-  }
-
-  private async getTTL(nodeHash) {
-    return await this.callMethod(this.readerContract, 'ttl', [nodeHash]);
-  }
-
-  /**
-   * This was done to make automated tests more configurable
-   */
-  async resolver(domain: string): Promise<string> {
-    const nodeHash = this.namehash(domain);
-    const resolverAddr = await this.callMethod(this.readerContract, 'resolver', [nodeHash]);
-    if (isNullAddress(resolverAddr)) {
-      throw new ResolutionError(ResolutionErrorCode.UnspecifiedResolver);
-    }
-    return resolverAddr;
-  }
-
-  /**
-   * This was done to make automated tests more configurable
-   */
-  private async getOwner(nodeHash) {
-    return await this.callMethod(this.readerContract, 'owner', [nodeHash]);
-  }
-
-  /**
-   * This was done to make automated tests more configurable
-   */
-  private async getResolutionInfo(domain: string) {
-    const nodeHash = this.namehash(domain);
-    return await Promise.all([
-      this.owner(domain),
-      this.getTTL(nodeHash),
-      this.resolver(domain),
-    ]);
-  }
-
-  protected getCoinType(currencyTicker: string): string {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    const constants: Bip44Constants[] = require('bip44-constants');
-    const coin = constants.findIndex(
-      item =>
-        item[1] === currencyTicker.toUpperCase() ||
-        item[2] === currencyTicker.toUpperCase(),
-    );
-    if (coin < 0 || !formatsByCoinType[coin]) {
-      throw new ResolutionError(ResolutionErrorCode.UnsupportedCurrency, {
-        currencyTicker,
-      });
-    }
-
-    return coin.toString();
-  }
+  private async callMethod(
+    contract: Contract,
+    method: string,
+    params: (string | string[])[],
+  ): Promise<any> {
+    try {
+      const result = await contract.call(method, params);
+      return result[0];
+    } catch (error) {
+      const { message }: { message: string } = error;
+      if (
+        message.match(/Invalid JSON RPC response/) ||
+        message.match(/legacy access request rate exceeded/)
+      ) {
+        throw new ResolutionError(ResolutionErrorCode.NamingServiceDown, {
+          method: this.name,
+        });
```

Comment:
This error isn't helpful at all: we make it harder both for our users to understand the problem and for us to debug it.